### PR TITLE
PSB-198 Load movie data 1 time into memory

### PR DESCRIPTION
**Pull request by aamster**
Modifies `FineTuning` class to load movie 1 time and pass to generator, to avoid loading for both train/test each time generator is instantiated.

### CODE REVIEW

File: deepinterpolation/cli/schemas.py [hunk 1]
```diff
@@ -722,6 +722,14 @@ class FineTuningSchema(argschema.schemas.DefaultSchema):
             'https://keras.io/api/models/model_training_apis/#fit-method')
     )
 
+    load_movie_into_memory = argschema.fields.Bool(
```
Comment (mikejhuang) [inline deepinterpolation/cli/schemas.py]: Naming it to `cache_data` would keep it consistent with the generator param with the same option.

https://github.com/AllenInstitute/deepinterpolation_ai_internal/blob/8fdfa16f8aa9aedd0da15dd3425bd31dba0f3d45/deepinterpolation/cli/schemas.py#L193

File: deepinterpolation/cli/fine_tuning.py [hunk 2]
```diff
@@ -126,6 +133,59 @@ def run(self):
             test_generator=data_test_generator
         )
 
+    def _maybe_load_movie_into_memory(
```
Comment (mikejhuang) [inline deepinterpolation/cli/fine_tuning.py]: nit: I don't see why this is named "maybe" load movie into memory. What's the maybe part of it? It loads it if the parameter `load_movie_into_memory` is True,

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import json
+
+import h5py
+import numpy as np
+import pytest
+
+from deepinterpolation.cli.schemas import GeneratorSchema
+
+
+@pytest.fixture
```
Comment (mikejhuang) [inline tests/conftest.py:10]: You mentioned previously you didn't like fixtures due to them being difficult to trace through while debugging. Why are you using them here? This fixture was only used for one test.

File: deepinterpolation/cli/fine_tuning.py [hunk 4]
```diff
@@ -126,6 +133,59 @@ def run(self):
...
+
+        if len(ophys_experiment_ids) > 1:
+            self.logger.warning(
+                'load_movie_into_memory will use a lot of '
+                'memory if there are multiple movies. Are you sure?')
+        for experiment_id in ophys_experiment_ids:
+            self.logger.info(f'Loading {experiment_id} into memory')
+            with h5py.File(
+                    train_data[experiment_id]["path"], "r") as f:
+                movs[experiment_id] = f['data'][()]
```
Comment (mikejhuang) [inline deepinterpolation/cli/fine_tuning.py:186]: This sets the keys of `movs` as `experiment_id`, whereas the class that it's used references the keys of `movs` as `video_index`.

Are `video_index` the same as `experiment_id`? This seems confusing to have two names referring to the same thing (if they are).

**Review by mikejhuang — APPROVED** [review]
Looks good if `experiment_id` in the way the keys to `mov` is set is the same as `video_index` in the way the keys to `mov` are gotten.

If not, please fix it before merging.

File: deepinterpolation/cli/fine_tuning.py (hunk 2 above)
Comment (aamster) [reply deepinterpolation/cli/fine_tuning.py]: That is the maybe part of it, if `load_movie_into_memory` is True

File: deepinterpolation/cli/fine_tuning.py (hunk 4 above)
Comment (aamster) [reply deepinterpolation/cli/fine_tuning.py:186]: `video_index` is what Jerome named it, it refers to the `experiment_id`

File: deepinterpolation/cli/schemas.py (hunk 1 above)
Comment (aamster) [reply deepinterpolation/cli/schemas.py]: i've renamed it

File: tests/conftest.py (hunk 3 above)
Comment (aamster) [reply tests/conftest.py:10]: It is being used by 2 tests. Didn't want to rewrite to use inheritance instead

**Pull request merged**